expo/expo
# [expo-go] Move home JS into apps/expo-go

# Why

This PR is a follow-up to https://github.com/expo/expo/pull/26130 and https://github.com/expo/expo/pull/26125 and moves the JS into the directory.

# How

git mv, grep for `home`, fix places that reference it.

# Test Plan

1. `yarn start` in `apps/expo-go`
2. Build android in android studio
3. Set DEV_KERNEL_SOURCE = LOCAL in EXBuildConstants.plist and build in xcode
4. See both locally build and use the local source.

# Checklist
